pages/articles.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+    <div class="pa-5">
+        <div v-if="loading">
+            <Loading :active="loading"
+                :can-cancel="false"
+                :is-full-page="true"
+                color="#007bff"
+            />
+        </div>
+        <div v-else>
+            <v-row align="center" class="mx-0">
+                <v-col :cols="4" v-for="article, index in currentArticleList" :key="index">
+                    <ArticleCard
+                        :key="index"
+                        :articleData="article"
+                    />
+                </v-col>
+            </v-row>
+
+            <v-pagination
+                circle
+                v-model="currentPage"
+                :length="totalPages"
+                @input="onPageNumberClick"
+            ></v-pagination>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+// Library imports
+import { Vue, Component } from 'vue-property-decorator'
+import Loading from 'vue-loading-overlay'
+import 'vue-loading-overlay/dist/vue-loading.css'
```

Comment:
I'm not aware of use case, but if you are showing loader on entire page then it would be easy to use loader in layout instead of repeating it again in components.

Reply:
I have added Loader component here to show the loader while the paginated list is getting loaded.
In real project we can make a generic global component and use it.